programs/vote/src/vote_state/handler.rs

```diff
@@ -1680,4 +1881,287 @@ mod tests {
             ExpectedVoteStateVersion::V3,
         );
     }
+
+    #[test]
+    fn test_deinitialize_vote_account_state_v4() {
+        let vote_pubkey = Pubkey::new_unique();
+        let rent = Rent::default();
+
+        // First create a vote account that's too small for v4.
+        let v1_14_11_size = VoteState1_14_11::size_of();
+        let lamports = rent.minimum_balance(v1_14_11_size);
+        let vote_account = AccountSharedData::new(lamports, v1_14_11_size, &id());
+
+        // Deinitialize - should fail.
+        deinit_vote_account_state_v4_and_assert(vote_pubkey, vote_account, rent.clone());
+
+        // Create a vote account that's too small for v4, but has enough
+        // lamports for resize.
+        let v4_size = VoteStateV4::size_of();
+        let lamports = rent.minimum_balance(v4_size);
+        let vote_account = AccountSharedData::new(lamports, v1_14_11_size, &id());
+
+        // Deinitialize - should resize and create v4.
+        deinit_vote_account_state_v4_and_assert(vote_pubkey, vote_account, rent.clone());
+
+        // Now create a vote account that's large enough for v4.
+        let lamports = rent.minimum_balance(v4_size);
+        let vote_account = AccountSharedData::new(lamports, v4_size, &id());
+
+        // Deinitialize - should create v4.
+        deinit_vote_account_state_v4_and_assert(vote_pubkey, vote_account, rent);
+    }
+
+    #[test]
+    fn test_v4_commission_basis_points() {
+        // Test that commission is properly converted to basis points
+        // (multiplied by 100), as per SIMD-0185.
+        let mut handler = VoteStateHandler::new_v4(VoteStateV4::default());
+
+        for (input, expected) in [
+            (0, 0),
+            (1, 100),
+            (5, 500),
+            (10, 1_000),
+            (25, 2_500),
+            (50, 5_000),
+            (75, 7_500),
+            (100, 10_000),
```

Comment:
commission can be set to 255 actually (no good reason for this) so if you want to test max value you should be 255